### PR TITLE
Add `trunkVersion` and `branchId` support

### LINKED ISSUE #6245: Collect should provide `trunkVersion` and `branchId` to servers
~~Blocked by #6267~~

So that servers are able to deal with conflicts from multiple clients performing offline updates, Collect should provide `trunkVersion` and `branchId` as part of entity submissions.

## Acceptance

Using XForms that include `branchId` as described in the [new spec](https://forum.getodk.org/t/form-spec-proposal-support-entity-updates-from-multiple-offline-clients/47890), the submission resulting from each of these actions satisfy the conditions in each of these scenarios:

- [x] Updating a locally created entity
     - New `branchId` (UUID)
     - `trunkVersion` is blank
     - `baseVersion` is the updated entity's version (which will be 1)
- [x] Updating an entity in the entity list (an "online" one)
     - New `branchId` (UUID)
     - `trunkVersion` is entity is the updated entity's version
     - `baseVersion` is entity is the updated entity's version
- [x] Updating a locally updated entity
     - Same `branchId` as the last update submission for this entity
     - Same `trunkVersion` as the last update submission for this entity
     - `baseVersion` is entity is the updated entity's version

Additionally, XForms that do not behave as the spec describes and set `baseVersion`, `branchId` or `trunkVersion` to unexpected values should not affect Collect. For example, if a form always sets `baseVersion` to "5" instead of to the provided `__version`, Collect would still increment `__version` in its local representation instead of always ending up with "6".

### PR DESCRIPTION
**Pull request by seadowg**
Closes #6245 
~~Blocked by #6267~~

As well as adding support for forms to receive `__trunkVersion` and `__branchId` and keeping track of these values correctly, this also changes how we increment `__version`. Before, Collect would parse the `baseVersion`, increment it and then store the result. This meant that a form might manipulate the `baseVersion` in an unexpected way which would Collect would honour. After chatting with @lognaturel, we've decided that this was actually an incorrect implementation of the following line in the spec:

> MUST increment the __version of its local entity representation by 1 when an update is successfully applied

Instead, we now increment our own representation of `__version` (`Entity#version`) and ignore the `baseVersion` in submissions.

I've also cleaned up how Collect treats forms that have both an `update` and a `create` action and added tests to define that behaviour.

#### Why is this the best possible solution? Were any other approaches considered?

Not a lot to discuss here! I'll put comments inline for anything that seems worth mentioning.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

This is mostly all additional, but there's a risk that changes to the way we increment version and to how we treat forms that have both an `update` and `create` action might alter how we expect Collect to behave. It'd be worth testing both of these out and checking we're happy with how everything works.

#### Before submitting this PR, please make sure you have:
- [x] added or modified tests for any new or changed behavior
- [x] run `./gradlew connectedAndroidTest` (or `./gradlew testLab`) and confirmed all checks still pass
- [x] added a comment above any new strings describing it for translators
- [x] added any new strings with date formatting to `DateFormatsTest`
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: entities/src/test/java/org/odk/collect/entities/LocalEntityUseCasesTest.kt [hunk 1]
```diff
@@ -19,20 +23,92 @@ class LocalEntityUseCasesTest {
...
+        val formEntities = EntitiesExtra(listOf(formEntity))
+
+        LocalEntityUseCases.updateLocalEntitiesFromForm(formEntities, entitiesRepository)
+        val entities = entitiesRepository.getEntities("things")
+        assertThat(entities.size, equalTo(1))
+        assertThat(entities[0].version, equalTo(2))
+    }
+
+    @Test
+    fun `updateLocalEntitiesFromForm updates properties on update`() {
```
Comment (grzesiek2010) [inline entities/src/test/java/org/odk/collect/entities/LocalEntityUseCasesTest.kt:64]: What kind of update is happening here?
You have properties `"prop" to "value"` in the original entity then you do the update and it's still the same.

**Review by grzesiek2010 — CHANGES_REQUESTED** [review]

File: entities/src/test/java/org/odk/collect/entities/LocalEntityUseCasesTest.kt (hunk 1 above)
Comment (seadowg) [reply entities/src/test/java/org/odk/collect/entities/LocalEntityUseCasesTest.kt:64]: Yeah, that test makes no sense 😂. Let me have another look!

**Review by grzesiek2010 — APPROVED** [review]

**Comment by WKobus** [comment]
**Tested with success**

Verified on device with Android 14

**Verified cases:**

- Create or update entity forms
- Regression check on entity forms
- Local entities disabled/enabled

**Comment by dbemke** [comment]
**Tested with success**

Verified on device with Android 10

**Pull request merged**